log4j-api/src/main/java/org/apache/logging/log4j/util/EnvironmentPropertySource.java

```diff
@@ -75,32 +65,17 @@ public CharSequence getNormalForm(final Iterable<? extends CharSequence> tokens)
 
 	@Override
 	public Collection<String> getPropertyNames() {
-		try {
 			return System.getenv().keySet();
-		} catch (final SecurityException e) {
-			logException(e);
-			return PropertySource.super.getPropertyNames();
-		}
```

Comment:
Same as above: the `getPropertyNames` is used for caching, but is not necessary for Log4j2.

I would leave the `catch`.